src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -359,6 +361,28 @@ public UUID getFirstSourceDatasetIdFromSnapshotId(
         .orElseThrow(() -> new DatasetNotFoundException("Source dataset for snapshot not found"));
   }
 
+  public SnapshotPreviewModel retrievePreview(UUID snapshotId, String tableName, Integer count) {
+    Snapshot snapshot = retrieve(snapshotId);
+
+    Optional<SnapshotTable> table =
+        snapshot.getTables().stream().filter(t -> t.getName().equals(tableName)).findFirst();
+
+    if (table.isEmpty()) {
+      throw new SnapshotPreviewException("No snapshot table exists with the name: " + tableName);
+    }
+
+    try {
+      List<Map<String, Object>> values = bigQueryPdao.getSnapshotTable(snapshot, tableName, count);
+
+      SnapshotPreviewModel previewModel = new SnapshotPreviewModel();
+      previewModel.setResult(Collections.singletonList(values));
```

Comment:
Not a huge deal but I think we should migrate our code to use the new immutable collection constructors instead of using `Collections`. They're nice in that it's the same form for no elements, one element, or many elements.
```suggestion
      previewModel.setResult(List.of(values));
```